airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresCatalogHelper.java

```diff
@@ -75,6 +75,19 @@ public static AirbyteStream setIncrementalToSourceDefined(final AirbyteStream st
     return stream;
   }
 
+  public static boolean isStreamAView(final JdbcDatabase database, final AirbyteStream stream) throws SQLException {
+    final String query = "SELECT COUNT(*) FROM pg_catalog.pg_views WHERE schemaname = ? AND viewname = ?";
+    final List<JsonNode> result = database.queryJsons(query, stream.getNamespace(), stream.getName());
+    return result.get(0).get("count").asInt() > 0;
+  }
+
+  public static AirbyteStream setFullRefreshToView(final JdbcDatabase database, final AirbyteStream stream) throws SQLException {
+    if (isStreamAView(database, stream)) {
```

Comment:
I think this means we will run this query N times, once for all N streams.  Could we load up the list of streams once instead?

Reply:
Great idea! I just pushed a commit that implements this idea. Please take another look, thanks!